feat #14 - 사용자 기능 구현

nswon opened the pull request:
### 변경사항
**AS-IS**
### 회원가입
- 계정 중복체크
- 비번 암호화
- 인증코드 전송

### 가입승인
- 계정, 비번, 인증코드 검사
- 가입승인상태로 변경

### 로그인
- 계정, 비번 검사
- JWT 토큰 발급

**TO-BE** 
- [x] IllegalArgumentException -> CustomException으로 변경
- [x] 비번 제약조건
- [ ] 테스트코드 (구현하다보니 테스트코드까지 하면 PR이 너무 커질 것 같아서, 따로 이슈를 파서 진행해도 될까요? 아니면 여기서 테스트코드까지 마무리하는게 좋을까요?)

### 테스트
- [ ] 테스트 코드
- [x] API 테스트

nswon committed:
feat #14 - 사용자 기능 구현
- 회원가입 흐름: 요청을 보내면  계정 중복체크와 비번 암호화를 하고,  Member 저장 후 인증코드 전송
- 가입승인 흐름: 계정, 비밀번호, 인증코드가 올바르면 가입승인 상태로 변경
- 로그인 흐름: 계정과 비번으로 로그인을 하면 유효성 체크 후 JWT 토큰 발급

nswon committed:
feat #14 - 회원가입시 비밀번호 제약조건 추가
- 비밀번호는 최소 10자 이상이어야 합니다.
- 숫자, 문자, 특수문자 중 2가지 이상을 포함해야 합니다.

nayoonk928 commented on src/main/java/com/wanted/teamV/controller/MemberController.java:
```diff
@@ -0,0 +1,43 @@
+    private final MemberService memberService;
+
+    @PostMapping
+    public ResponseEntity<MemberCodeResDto> join(@RequestBody MemberJoinReqDto memberJoinReqDto) {
```
Controller 클래스에서는 매개변수 작성할 때 아래와 같이 수정해 주시면 좋을 것 같아요.
```
public ResponseEntity<> join(
       @RequestBody MemberJoinReqDto memberJoinReqDto
) {
     ....
}
```

nswon replied:
수정했습니다 👍

nayoonk928 commented on src/main/java/com/wanted/teamV/entity/Member.java:
```diff
@@ -39,4 +42,14 @@ public Member(String account, String password, String email, String code, Member
+
+    public void verifyCode(String code) {
+         if(!this.code.equalsIgnoreCase(code)) {
+            throw new CustomException(INVALID_AUTHENTICATION_CODE, INVALID_AUTHENTICATION_CODE.getMessage(), INVALID_AUTHENTICATION_CODE.getStatus());
```
`throw new CustomException(INVALID_AUTHENTICATION_CODE);`
위 처럼 작성하셔도 message와 status 모두 반환되니 참고해주세요.

nswon replied:
오 어떻게 써야할지 몰라서 저렇게 썼었는데, 고치도록 하겠습니다

nayoonk928 commented on src/main/java/com/wanted/teamV/repository/MemberRepository.java:
```diff
@@ -1,11 +1,20 @@
 
+    default Member getByAccount(String account) {
+        return findByAccount(account)
+                .orElseThrow(() -> new CustomException(NOT_FOUND, NOT_FOUND.getMessage(), NOT_FOUND.getStatus()));
```
이 부분도 NOT_FOUND만 전달해도 괜찮습니다!
다른 파일들의 new CustomException 부분도 ErrorCode 만 전달해주셔도 됩니다.

nswon replied:
수정 완료했습니다!

nayoonk928 reviewed (commented):
코드 잘 봤습니다! record 사용한 코드 처음 보는데 깔끔하고 좋은 것 같아요.
CustomException 사용한 부분과 Controller 코드 스타일만 확인해 주세요.

nswon committed:
feat #14 - 컨벤션 적용
- Controller에서 매개변수 작성시 컨벤션 적용
- IllegalArgumentException -> CustomException으로 변경
- CustomException 처리방식 수정
